install.sh

```diff
@@ -961,11 +961,16 @@ EOF
 
 # --- write systemd or openrc service file ---
 create_service_file() {
-    [ "${HAS_SYSTEMD}" = true ] && create_systemd_service_file
+    [ "${HAS_SYSTEMD}" = true ] && create_systemd_service_file && restore_systemd_service_file_context
```

Comment:
Would this perhaps fit better with the rest of the context changes? It doesn't feel right just blindly calling restorecon when creating the systemd unit.

https://github.com/k3s-io/k3s/blob/7d38b4a3dbd335c155836c4413987589a64e6f3c/install.sh#L560-L569

Reply:
unfortunately the systemd unit file deployment happens after this section in the create_service_file function, and a chicken-egg problem happens because the context is applied before the systemd unit file is deployed
```
setup_selinux
...    
create_env_file
create_service_file
```

Reply:
My only concern is that we're going to get a request to disable this behavior. Whenever we add something to the install script, someone always asks for an option to not do it.